### ISSUE TITLE
Malformed ECDHE parameters from client may cause panic

### ISSUE THREAD

**Issue by briansmith** [ctz/rustls#641]
I didn't verify this experimentally; I just read the code:

```
    fn decode_client_params(&self, kx_params: &[u8]) -> Option<ClientECDHParams> {
        let mut rd = Reader::init(kx_params);
        let ecdh_params = ClientECDHParams::read(&mut rd).unwrap();
        if rd.any_left() {
            None
        } else {
            Some(ecdh_params)
        }
    }
```
It seems like the `unwrap()` is reachable, if there's a syntax error in the ClientECDHParams. Either way, it seems like the `unwrap()` can easily be replaced with `?`.

**Comment by djc** [comment]
Fixed that too in #644.

**Comment by ctz** [comment]
Released 0.19.1 with this backported.
